proc_dash/app.py

```diff
@@ -474,18 +562,25 @@ def generate_overview_status_fig_for_participants(parsed_data):
     Input(
         "interactive-datatable", "data"
     ),  # Input not triggered by datatable frontend filtering
+    State("memory-pipelines", "data"),
     prevent_initial_call=True,
 )
-def update_overview_status_fig_for_records(data):
+def update_overview_status_fig_for_records(data, pipelines_dict):
     """
     When visible data in the overview datatable is updated (excluding built-in frontend datatable filtering
-    but including component filtering for multiple sessions), generate stacked bar plot of pipeline_complete
-    statuses aggregated by pipeline. Counts of statuses in plot thus correspond to unique records (unique
-    participant-session combinations).
+    but including custom component filtering), generate stacked bar plot of pipeline_complete statuses aggregated
+    by pipeline. Counts of statuses in plot thus correspond to unique records (unique participant-session
+    combinations).
     """
     if data is not None:
-        return plot.plot_pipeline_status_by_records(
-            pd.DataFrame.from_dict(data)
+        data_df = pd.DataFrame.from_dict(data)
```

Comment:
I think this is getting a bit hard to read. :cherries:: maybe using the full if-else syntax here would make it visually clearer how you are using all of these return routes